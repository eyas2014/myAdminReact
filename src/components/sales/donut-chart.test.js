import React from 'react';
import DonutChart from './donut-chart';
import enzyme from 'enzyme';

var data={ colors: ["#3c8dbc", "#f56954", "#00a65a"],
			   data: [{label: "Download Sales", value: 12},
				  {label: "In-Store Sales", value: 30},
				  {label: "Mail-Order Sales", value: 20}]
			}


document.body.innerHTML='<div  id="root"></div>';

jest.mock('../../library/morris/morris.js');

const wrapper=enzyme.mount(<DonutChart id="sales-chart"
								colors= {data.colors}
								data= {data.data}/>, 
							{attachTo: document.getElementById('root')});


it ('should match the snapshot', ()=>{
   expect($('#root').html()).toMatchSnapshot();
})

it ('should call Morris with listed args', ()=>{
	expect(require('../../library/morris/morris.js').isCalledWith).toEqual(
		{	resize: true,
			element: "sales-chart",
			data: [{label: "Download Sales", value: 12},
				  {label: "In-Store Sales", value: 30},
				  {label: "Mail-Order Sales", value: 20}],
			colors: ["#3c8dbc", "#f56954", "#00a65a"],
	    	hideHover:'auto'});
})
