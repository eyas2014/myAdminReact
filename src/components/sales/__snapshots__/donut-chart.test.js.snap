// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `"<div class=\\"chart tab-pane active\\" id=\\"sales-chart\\" style=\\"position: relative; height: 300px;\\"></div>"`;
